src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs

```diff
@@ -64,15 +65,14 @@ public async Task<IFileStoreEntry> GetFileInfoAsync(string path)
             {
                 var blob = GetBlobReference(path);
 
-                if (!await blob.ExistsAsync())
-                {
-                    return null;
-                }
-
                 var properties = await blob.GetPropertiesAsync();
 
                 return new BlobFile(path, properties.Value.ContentLength, properties.Value.LastModified);
             }
+            catch (RequestFailedException ex) when (ex.ErrorCode == "BlobNotFound") // Instead of ExistsAsync() check which is 'slow' if we're expecting to find the blob 
```

Comment:
Does it work with this property instead of a string: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/src/Generated/Models/BlobErrorCode.cs#L230

Reply:
Yes, tested it out and using the BlobErrorCode works.